File: js/stack/150evalRPN.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
	const SIGN = {
		'*' : (a, b) => a * b,
		'/' : (a, b) => Math.trunc(a / b),
		'+' : (a, b) => a + b,
		'-' : (a, b) => a - b
	}
	let stack = [];
	let flg = '+-*/';
	for (let tok of tokens) {
		if(tok in SIGN){
			let right = parseInt(stack.pop());
			let left = parseInt(stack.pop());
			let res = SIGN[tok](left,right);
			stack.push(res);			
		}else{
			stack.push(tok);
		}		
	}
	return stack.pop();
};

let testcase = evalRPN(["2","1","+","3","*"]);
console.log(testcase);
export default evalRPN;

File: py/stack/150evalRPN.py
import operator
from typing import List


class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        SIGN = {'+':operator.add,'-':operator.sub,'*':operator.mul,'/':operator.truediv}
        stack = []
        for tok in tokens:
            if tok not in SIGN:
                stack.append(tok)
            else:
                b = stack.pop()
                a = stack.pop()
                stack.append(int(SIGN[tok](int(a),int(b))))
        return int(stack[0])

solution = Solution()
ans = solution.evalRPN(["2","1","+","3","*"])
print(ans)